main.cpp

```diff
@@ -296,13 +200,20 @@ struct FloatType
     FloatType& multiply(float rhs);
     FloatType& divide(float rhs);
 
+    FloatType& pow(float rhs);
+    FloatType& pow(const FloatType& rhs);
+    FloatType& pow(const DoubleType& rhs);
+    FloatType& pow(const IntType& rhs);
+
 private:
+    FloatType& powInternal(float* base, float exp);
```

Comment:
The first argument is not needed.
```
 2) your powInternal() function should do something like this in its body:    *val = std::pow( *val, arg );
```
the internal `val/value` pointer is always used inside this function, and does not need to be passed as a function argument
- [ ] revise